core/src/internal_flags.rs

```diff
@@ -0,0 +1,84 @@
+//! Utilities for and tracking of internal versions which alter history in incompatible ways
+//! so that we can use older code paths for workflows executed on older core versions.
+
+use std::collections::{BTreeSet, HashSet};
+use temporal_sdk_core_protos::temporal::api::{
+    history::v1::WorkflowTaskCompletedEventAttributes, sdk::v1::WorkflowTaskCompletedMetadata,
+};
+
+/// This enumeration contains internal flags that may result in incompatible history changes with
+/// older workflows, or other breaking changes.
+///
+/// When a flag has existed long enough the version it was introduced in is no longer supported, it
+/// may be removed from the enum. *Importantly*, all variants must be given explicit values, such
+/// that removing older variants does not create any change in existing values. Removed flag
+/// variants must be reserved forever (a-la protobuf), and should be called out in a comment.
+#[repr(u32)]
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Copy, Clone, Debug)]
+pub(crate) enum CoreInternalFlags {
+    /// In this flag, additional checks were added to a number of state machines to ensure that
+    /// the ID and type of activities, local activities, and child workflows match during replay.
+    IdAndTypeDeterminismChecks = 0,
```

Comment:
To confirm, basically when this version of core is incorporated into the SDK, we are gonna add this flag to the next wft completion for every workflow using the new SDK?

I wonder if you want to pack a few more features in here pre-launch, e.g. the one for patch search attr or any others. I guess lang can choose when to update core and the next wft complete at that point will have all of them, so not in this PR necessarily.

Reply:
No, only the first WFT in which the check is used, so it's easy to add them whenever needed, the comment in the upstream proto describes the precise behavior.

Reply:
> only the first WFT in which the check is used

Ok, I think I have it now. But I am still confused about something.

Do you ever check ID/type mismatches during non-replay? How is it possible to check during non-replay when you don't even have the event to check against? Basically, when would `replaying` ever be false for `try_use` when you're testing against an event during replay? (sorry, reading code but not clear)

The reason I ask is because I want to confirm the "replaying" bool when this flag applies to confirm it won't do an ID/type for older tasks (which obviously it is built not to, I just need understanding).